addons/account/tests/test_account_move_reconcile.py

```diff
@@ -4000,6 +4000,104 @@ def test_reconcile_cash_basis_tax_grid_multi_taxes(self):
             {'debit': 0.0,      'credit': 0.01,     'tax_ids': [],              'tax_tag_ids': self.tax_tags[5].ids,    'account_id': self.tax_account_2.id},
         ])
 
+    def test_matching_number_full_reconcile(self):
+        currency = self.env.company.currency_id
+        line_a = self.create_line_for_reconciliation(1000, 1000, currency, '2016-01-01')
+        line_b = self.create_line_for_reconciliation(-1000, -1000, currency, '2016-01-01')
+        (line_a + line_b).reconcile()
+        self.assertFullReconcile(line_a.full_reconcile_id, (line_a + line_b))
+        self.assertEqual(line_a.matching_number, str(line_a.full_reconcile_id.id))
+        self.assertEqual(line_a.matching_number, line_b.matching_number)
+
+    def test_matching_number_partial_single_reconcile(self):
+        currency = self.env.company.currency_id
+        line_a = self.create_line_for_reconciliation(1000, 1000, currency, '2016-01-01')
+        line_b = self.create_line_for_reconciliation(-500, -500, currency, '2016-01-01')
+        (line_a + line_b).reconcile()
+        self.assertEqual(line_a.matching_number, f'P{line_a.matched_credit_ids.id}')
+        self.assertEqual(line_a.matching_number, line_b.matching_number)
+
+    def test_matching_number_partial_multi_reconcile(self):
+        currency = self.env.company.currency_id
+        line_a = self.create_line_for_reconciliation(1000, 1000, currency, '2016-01-01')
+        line_b = self.create_line_for_reconciliation(-500, -500, currency, '2016-01-01')
+        line_c = self.create_line_for_reconciliation(-1000, -1000, currency, '2016-01-01')
+        line_d = self.create_line_for_reconciliation(1000, 1000, currency, '2016-01-01')
+        (line_a + line_b).reconcile()
+        (line_a + line_c).reconcile()
+        (line_c + line_d).reconcile()
+        self.assertEqual(line_a.matching_number, f'P{line_a.matched_credit_ids.ids[0]}')
+        self.assertEqual(line_b.matching_number, line_a.matching_number)
+        self.assertEqual(line_c.matching_number, line_a.matching_number)
+        self.assertEqual(line_d.matching_number, line_a.matching_number)
+
+        line_b.remove_move_reconcile()
+        self.assertEqual(line_a.matching_number, f'P{line_a.matched_credit_ids.ids[0]}')
+        self.assertEqual(line_b.matching_number, False)
+        self.assertEqual(line_c.matching_number, f'P{line_c.matched_debit_ids.ids[0]}')
+        self.assertEqual(line_d.matching_number, line_c.matching_number)
```

Comment:
there's something odd: why don't we have `self.assertEqual(line_a.matching_number, line_c.matching_number) `? They all had the same reconciliation ID, a partial is removed, name _can_ change since the min isn't forcefully the same ID anymore, but they still should have the same reconciliation ID

Also, what happens in case the removed partial isn't the `min`, are they still linked altogether under the same name? (that's what I'm expecting)

Reply:
I persist: `a`, `c` and `d` are still all linked together, so they should get the same ID.

Imagine that c isn't -1000 but -100, don't you agree that you'd like the following 2 use cases to have the exact same outcome?

>  1)    (line_a + line_c).reconcile()
>         (line_c + line_d).reconcile()

>  2)    (line_a + line_b).reconcile()
>         (line_a + line_c).reconcile()
>         (line_c + line_d).reconcile()
>         line_b.remove_move_reconcile()

Reply:
> Oh yes 🤦 They actually all have the same number since `line_a.matched_credit_ids.ids[0] == line_c.matched_debit_ids.ids[0]`
> 
> > Also, what happens in case the removed partial isn't the min, are they still linked altogether under the same name? (that's what I'm expecting)
> 
> Yes

ookay, that's reassuring :D

Reply:
Should I force push that now? (I'm not sure because this PR already have a r+)

Reply:
Let's fix these details after the merge :)